doc/api/quic.md

```diff
@@ -0,0 +1,2276 @@
+
+#### quicsession.authenticated
+<!--YAML
+added: REPLACEME
+-->
+* Type: {boolean}
+
+True if the certificate provided by the peer during the TLS 1.3
+handshake has been verified.
+
+#### quicsession.authenticationError
+
+* Type: {Object} An error object
+
+If `quicsession.authenticated` is false, returns an `Error` object
+representing the reason the peer certificate verification failed.
+
+#### quicsession.bidiStreamCount
+<!-- YAML
+added: REPLACEME
+-->
+
+* Type: {BigInt}
+
+A `BigInt` representing the total number of bidirectional streams
+created for this `QuicSession`.
+
+#### quicsession.blockCount
+<!-- YAML
+added: REPLACEME
+-->
+
+* Type: {BigInt}
+
+A `BigInt` representing the total number of times the `QuicSession` has
+been blocked from sending stream data due to flow control.
+
+Such blocks indicate that transmitted stream data is not being consumed
+quickly enough by the connected peer.
+
+#### quicsession.bytesInFlight
+<!-- YAML
+added: REPLACEME
+-->
+
+* Type: {number}
+
+The total number of unacknowledged bytes this QUIC endpoint has transmitted
+to the connected peer.
+
+#### quicsession.bytesReceived
+<!-- YAML
+added: REPLACEME
+-->
+
+* Type: {BigInt}
+
+A `BigInt` representing the total number of bytes received from the peer.
+
+#### quicsession.bytesSent
+<!-- YAML
+added: REPLACEME
+-->
+
+* Type: {BigInt}
+
+A `BigInt` representing the total number of bytes sent to the peer.
+
+#### quicsession.cipher
+<!-- YAML
+added: REPLACEME
+-->
+
+* Type: {Object}
+  * `name` {string} The cipher algorithm name.
+  * `type` {string} The TLS version (currently always `'TLSv1.3'`).
+
+Information about the cipher algorithm selected for the session.
+
+#### quicsession.close(\[callback\])
+<!-- YAML
+added: REPLACEME
+-->
+
+* `callback` {Function} Callback invoked when the close operation is completed
+
+Begins a graceful close of the `QuicSession`. Existing `QuicStream` instances
+will be permitted to close naturally. New `QuicStream` instances will not be
+permitted. Once all `QuicStream` instances have closed, the `QuicSession`
+instance will be destroyed.
+
+#### quicsession.closeCode
+<!-- YAML
+added: REPLACEME
+-->
+* Type: {Object}
+  * `code` {number} The error code reported when the `QuicSession` closed.
+  * `family` {number} The type of error code reported (`0` indicates a QUIC
+    protocol level error, `1` indicates a TLS error, `2` represents an
+    application level error.)
+
+#### quicsession.closing
+<!-- YAML
+added: REPLACEME
+-->
+
+* Type: {boolean}
+
+Set to `true` if the `QuicSession` is in the process of a graceful shutdown.
+
+#### quicsession.destroy(\[error\])
+<!-- YAML
+added: REPLACEME
+-->
+
+* `error` {any}
+
+Destroys the `QuicSession` immediately causing the `close` event to be emitted.
+If `error` is not `undefined`, the `error` event will be emitted immediately
+before the `close` event.
+
+Any `QuicStream` instances that are still opened will be abruptly closed.
```

Comment:
should describe what happens to pending data (probably discard?) --- this is as far as I can get today in review.

Reply:
+1 ... yeah, it's is discarded. Specifically, pending writes are canceled.